src/dict/stardict.cc

```diff
@@ -525,29 +525,28 @@ string StardictDictionary::handleResource( char type, char const * resource, siz
 
         QString src = match.captured( 2 );
 
-        if ( src.indexOf( "://" ) >= 0 )
+        if ( src.indexOf( "://" ) >= 0 ) {
           articleNewText += match.captured();
-
+        }
         else {
-          std::string href = "\"gdau://" + getId() + "/" + src.toUtf8().data() + "\"";
-          QString newTag   = QString::fromUtf8(
-            ( addAudioLink( href, getId() ) + "<span class=\"sdict_h_wav\"><a href=" + href + ">" ).c_str() );
-          newTag += match.captured( 4 );
-          if ( match.captured( 4 ).indexOf( "<img " ) < 0 )
-
+          std::string href   = "\"gdau://" + getId() + "/" + src.toUtf8().data() + "\"";
+          std::string newTag = addAudioLink( href, getId() ) + "<span class=\"sdict_h_wav\"><a href=" + href + ">";
+          newTag += match.captured( 4 ).toUtf8().constData();
+          if ( match.captured( 4 ).indexOf( "<img " ) < 0 ) {
             newTag += R"( <img src="qrc:///icons/playsound.png" border="0" alt="Play">)";
+          }
           newTag += "</a></span>";
 
-          articleNewText += newTag;
+          articleNewText += QString::fromStdString( newTag );
         }
       }
       if ( pos ) {
         articleNewText += articleText.mid( pos );
         articleText = articleNewText;
         articleNewText.clear();
       }
-
-      return articleText.toStdString();
+      auto text = articleText.toUtf8();
+      return text.data();
     }
```

Comment:
How can this fix anything?

There isn't much difference.
* in the new code, `std::string(toUtf8().data()) `will seek for the null byte again 
* in the old code, toStdString's `std::string(toUtf8().data(),toUtf8().size())` will use a articleText's actual length

Should a comment be added states that article Text may contain null byte, and we want cuf off earlier?

Reply:
the original code is 
`return (articleText.toUtf8().data());`
I guess `articleText.toUtf8()` will generate a temporary string .  when chained with  `.data()`  .   sometimes the temporary variable has no longer existed.

Reply:
> * in the old code, toStdString's `std::string(toUtf8().data(),toUtf8().size())` will use the articleText's actual length

the old code is `return (articleText.toUtf8.data());`  ,`toStdString()` is my first attempt.

Reply:
https://bugreports.qt.io/browse/QTBUG-63274

Reply:
Any relation?
![image](https://github.com/user-attachments/assets/5d8df8fb-6ea5-499d-b4e1-cd5514467ff4)

https://doc.qt.io/qt-6/qstring.html#details:~:text=Note%3A%20Due,for%20more%20information.

Reply:
If a temporal toStdString at return position would crash, then there are various other places would also crash already, so I don't think this is the root problem.

:sweat_smile: I have no other ideas as of now.

Reply:
> Any relation?

I don't think so. That section of doc only says iterator or "pointer to the data of QString" may become invalidate.

But we don't use any of them here.

Reply:
I also can not be sure.

But Qt+MSVC  QString   are more likely to have issue as the https://bugreports.qt.io/browse/QTBUG-63274 implies

Reply:
This seems to be the correct way to set /MD /MT
```
set_property(TARGET foo PROPERTY
  MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
```
https://github.com/qt/qtbase/blob/02cb165ef8050230b477358e4136e9f0acd83eb6/cmake/QtPublicTargetHelpers.cmake#L395-L397

However, CMake's doc says 

> If the property is not set, then CMake uses the default value MultiThreaded$<$<CONFIG:Debug>:Debug>DLL to select a MSVC runtime library.

So, I suppose it is set correctly by CMake already.

https://cmake.org/cmake/help/latest/prop_tgt/MSVC_RUNTIME_LIBRARY.html

Reply:
should cmake_policy be set?
https://cmake.org/cmake/help/latest/prop_tgt/MSVC_RUNTIME_LIBRARY.html#:~:text=Note%20This%20property%20has%20effect%20only%20when%20policy%20CMP0091%20is%20set%20to%20NEW%20prior%20to%20the%20first%20project()%20or%20enable_language()%20command%20that%20enables%20a%20language%20using%20a%20compiler%20targeting%20the%20MSVC%20ABI.

Reply:
No,`cmake_minimum_required` will set policy version, and we currently set it aggressively high :sweat_smile:.

https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#policy-settings

https://github.com/xiaoyifang/goldendict-ng/blob/3ef35daec3fa7be0597255102e5db2288023e809/CMakeLists.txt#L1

Reply:
I think search in the `build.ninja` file in the build folder to confirm /MD /MT is set.